pkg/engine/internal/workflow/workflow.go

```diff
@@ -133,19 +137,42 @@ func (wf *Workflow) Run(ctx context.Context) (pipeline executor.Pipeline, err er
 		}
 	}
 
-	// TODO(rfratto): For logs queries, we want a system to limit how many scan
-	// tasks get sent to the runner at once.
-	//
-	// This will limit unnecessary resource consumption of workflows when
-	// there's a lot of compute capacity.
-	if err := wf.runner.Start(ctx, wrappedHandler, tasks...); err != nil {
-		pipeline.Close()
+	g := wf.dispatchTasks(ctx, wrappedHandler, tasks)
+	if err := g.Wait(); err != nil {
+		wrapped.Close()
 		return nil, err
 	}
 
 	return wrapped, nil
 }
 
+// dispatchTasks groups the slice of tasks by their associated "admission lane" (token bucket)
+// and dispatches them to the runner.
+// Tasks from different admission lanes are dispatched concurrently.
+// The caller needs to wait on the returned error group.
+func (wf *Workflow) dispatchTasks(ctx context.Context, handler TaskEventHandler, tasks []*Task) *errgroup.Group {
+	// TODO(chaudum): Make the capacity of the admission lanes configurable
+	wf.admissionControl = newAdmissionControl(ctx, defaultAdmissionControlOpts)
+
+	g, _ := errgroup.WithContext(ctx)
```

Comment:
The idea of `Start` taking in a slice of tasks was to try to avoid the overhead of locking the mutex too many times, especially since workflows may create quite a large number of tasks. 

What would it take to redesign this to allow us to send a batch of tasks to schedule? (It's ok if it's one batch per lane). 

If it's going to be too much work, I'm happy to keep this as-is and revisit once we get some stats for how what impact the scheduler's mutex has at scale.